src/signer.ts

```diff
@@ -0,0 +1,100 @@
+import { NullSigner } from "./exceptions";
+import {
+    solG2,
+    Domain,
+    getPubkey,
+    g2ToHex,
+    sign,
+    g1ToHex,
+    aggregateRaw,
+    solG1,
+    SecretKey,
+    randFr,
+    PublicKey,
+    parseG1,
+    parseG2,
+    verifyRaw,
+    verifyMultipleRaw,
+    hashToPoint,
+    init,
+} from "./mcl";
+
+export interface BlsSignerInterface {
+    pubkey: solG2;
+    sign(message: string): solG1;
+    verify(signature: solG1, pubkey: solG2, message: string): boolean;
+    verifyMultiple(
+        aggSignature: solG1,
+        pubkeys: solG2[],
+        messages: string[]
+    ): boolean;
+}
+
+// Useful when your real signer is not loaded but need a placeholder
+export class NullBlsSinger implements BlsSignerInterface {
+    get pubkey(): solG2 {
+        throw new NullSigner("NullSinger has no public key");
+    }
+    sign(message: string): solG1 {
+        throw new NullSigner("NullSinger dosen't sign");
+    }
+    verify(signature: solG1, pubkey: solG2, message: string): boolean {
+        throw new NullSigner("NullSinger dosen't verify");
+    }
+    verifyMultiple(
+        aggSignature: solG1,
+        pubkeys: solG2[],
+        messages: string[]
+    ): boolean {
+        throw new NullSigner("NullSinger dosen't verify");
+    }
+}
+
+export const nullBlsSigner = new NullBlsSinger();
+
+export class BlsVerifier {
+    constructor(public domain: Domain) {}
+    public verify(signature: solG1, pubkey: solG2, message: string) {
+        const signatureG1 = parseG1(signature);
+        const pubkeyG2 = parseG2(pubkey);
+        const messagePoint = hashToPoint(message, this.domain);
+        return verifyRaw(signatureG1, pubkeyG2, messagePoint);
+    }
+    public verifyMultiple(
+        aggSignature: solG1,
+        pubkeys: solG2[],
+        messages: string[]
+    ) {
+        const signatureG1 = parseG1(aggSignature);
+        const pubkeyG2s = pubkeys.map(parseG2);
+        const messagePoints = messages.map((m) => hashToPoint(m, this.domain));
+        return verifyMultipleRaw(signatureG1, pubkeyG2s, messagePoints);
+    }
+}
+
+export class BlsSigner extends BlsVerifier implements BlsSignerInterface {
+    static async getSigner(domain: Domain) {
+        await init();
+        const secret = randFr();
```

Comment:
Could be interesting to have the secret as a parameter to the function with the default value of randFr(). That way you preserve the current behaviour but also allow users to use the signer with a specified key pair